prow/tide/tide.go

```diff
@@ -1124,18 +1125,14 @@ func (c *changedFilesAgent) prChanges(pr *PullRequest) config.ChangedFilesProvid
 		}
 		c.RUnlock()
 
-		// We need to query the changes from GitHub.
-		changes, err := c.ghc.GetPullRequestChanges(
-			string(pr.Repository.Owner.Login),
-			string(pr.Repository.Name),
-			int(pr.Number),
-		)
+		cloneRef := fmt.Sprintf("%s/%s", org, repo)
+		r, err := c.gc.Clone(cloneRef)
```

Comment:
We probably shouldn't clone the repo for every PR, every sync loop. That sounds prohibitively expensive for large repos with lots of PRs (e.g. `k/k` after code thaw). Can we instead check out each repo at most once per sync loop and reuse the checkout to get the diffs for each PR?

Reply:
Thank you for the review. I will update the PR.